test/web/RawSocketHost.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using System.Net;
+using System.Net.Sockets;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+
+using Socket = Tmds.LinuxAsync.Socket;
+
+namespace web
+{
+    public class RawSocketHost
+    {
+        private const int BufferSize = 512;
+        private const string Response =
+            "HTTP/1.1 200 OK\r\nDate: Tue, 31 Mar 2020 14:49:06 GMT\r\nContent-Type: application/json\r\nServer: Kestrel\r\nContent-Length: 27\r\n\r\n{\"message\":\"Hello, World!\"}";
+
+        private static ReadOnlySpan<byte> RequestEnd => new byte[] {13, 10, 13, 10}; // "\r\n\r\n"
+        
+        private string[] _args;
+        private readonly CommandLineOptions _options;
+
+        private IPEndPoint _serverEndpoint;
+
+        public RawSocketHost(CommandLineOptions options, string[] args)
+        {
+            _options = options;
+            _args = args;
+            
+            ConfigurationBuilder bld = new ConfigurationBuilder();
+            bld.AddCommandLine(_args);
+            var cfg = bld.Build();
+            string url = cfg["urls"];
+
+            string[] data = url.Substring(7).Split(':');
+            IPAddress ip = IPAddress.Parse(data[0]);
+            int port = int.Parse(data[1]);
+
+            _serverEndpoint = new IPEndPoint(ip, port);
+        }
+
+        public void Run()
+        {
+            RunAsync().GetAwaiter().GetResult();
+        }
+
+        public async Task RunAsync()
+        {
+            
+            using Socket listener = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+            listener.Bind(_serverEndpoint);
+            listener.Listen(1024);
+            Console.WriteLine($"Raw server listening on {_serverEndpoint}");
+            
+            byte[] receiveBuffer = new byte[BufferSize];
+            byte[] sendBuffer = Encoding.ASCII.GetBytes(Response);
+
+            while (true)
+            {
+                using Socket handler = await listener.AcceptAsync();
+
+                while (true)
```

Comment:
Can you add a class that handles a single client?
```cs
Socket clientSocket = await listener.AcceptAsync();
new ClientConnection(clientSocket, <options>).HandleClient();
```

`ClientConnection` should use 2 `SocketAsyncEventArgs`, one for receiving, and one for sending.
We can support some of the options, like `c=inline`, by setting properties on the `SocketAsyncEventArgs`.